src/data.ml

```diff
@@ -181,6 +181,7 @@ and type t = (clause * int) list = struct
   type t = elt list [@@deriving show]
   
   let get_time_stamp = snd
+  let equal a b = a = b
```

Comment:
I think you should use the timestamp as a unique id, here you are comparing the clause "code" which can be large.

Reply:
I totally agree